pkg/app.go

```diff
@@ -48,6 +48,7 @@ func init() {
 type App struct {
```

Comment:
> I think we can reuse this structure. "Multiple services" => "Multiple App". Although it looks a little bit strange, I think it's much easier.

Do you mean, instead of having the `Service` abstraction we just somehow create multiple apps? like so:
```
package main

func main() {
  app := beego.NewApp()
  app.Router("/login", &MainController{})
  app.Run()

  app2 := beego.NewApp()
  app2.Router("/foo", &FooController{})
  app2.Run()
```
I haven't been around the code for long, so I'm just guessing here.
Wouldn't that change the basic interface of how beego works presently? How would `BConfig` work in this case? Since now there is no default service running or something?

Or are you proposing that we add a `[]App` field in `type App struct` ? And do the same thing we're doing with `Service`?

Reply:
Using `Service` abstraction is a good idea. 

I prefer that:
```go
type Service struct {
    Handlers *ControllerRegister
    Server   *http.Server
    Config *Config // this field has the same type of BConfig. 
}
```
Do you notice that `BConfig` is an instance `Config`?

![image](https://user-images.githubusercontent.com/9923838/91060367-70f9fd80-e65d-11ea-97d5-19ba11f1ab47.png)

The key point is that `Service` uses its own `Config` rather than `BConfig`.

And when we create `Service`, we do something like that:
```
// this is similar with App
// I think BConfig becoming the global configuration for all service instances.
service1 := NewServiceWithGlobalConfig(BConfig)

service1.Router(xxx)

// set some configuration items which is different from global configuration.
// service1.Config.xxx= xxx


// or  service1.Config.port = "8080"
// beego.Run(), it means that App has all service instances.
service1.Run("localhost:8080")

// ...

func NewServiceWithConfig(Config *Config) *Service {
    s := &Service{}
    s.Config = Config
    return s
}

// this is similar with App.Run
(s *Service) Run(uri string) {
   s.Config.Port = parsePort()
}

```

So, in fact, I think `current App stucture` == `Service in your definition`

This is just a proposal. 😄